### PR TITLE
feat: Creating structure and providing simple test for Blockly automated functional test

### PR DESCRIPTION
**Pull request by ericblackmonGoogle**
## The basics

- [ x] I branched from develop
- [ x] My pull request is against develop
- [x ] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x ] I ran `npm run format` and `npm run lint`

### Resolves

Creates the configuration for our automated functional testing and also provides a small example.

### Proposed Changes

Creates the configuration for our automated functional testing and also provides a small example.

#### Behavior Before Change

No automated functional test run along side of the test suite

#### Behavior After Change

The structure for automated test now run along side of the test suite

### Reason for Changes

To allow our team to begin running automated functional test

### Test Coverage

It is a test

### Documentation

A document detailing the test we would like to run will be done by mid next week
### Additional Information

N/A

### CODE REVIEW

File: tests/browser/test/basic_block_factory_test.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      options.capabilities['goog:chromeOptions'].args.push(
+          '--headless', '--no-sandbox', '--disable-dev-shm-usage',);
+    } else {
+      // --disable-gpu is needed to prevent Chrome from hanging on Linux with
+      // NVIDIA drivers older than v295.20. See
+      // https://github.com/google/blockly/issues/5345 for details.
+      options.capabilities['goog:chromeOptions'].args.push('--disable-gpu');
+    }
+  // Use Selenium to bring up the page
+  const url = 'https://blockly-demo.appspot.com/static/demos/blockfactory/index.html';
```
[the remark this answers is not in the document]
Comment (ericblackmonGoogle) [reply tests/browser/test/basic_block_factory_test.js:44]: Yep we can, do we have any local versions for blockfactory that I can substitute?

[the remark this answers is not in the document]
Comment (BeksOmega) [reply tests/browser/test/basic_block_factory_test.js]: Yep! The block factory lives [here](https://github.com/google/blockly/tree/develop/demos/blockfactory).

But I'm realizing this relies on the compressed files, and I'm not sure if we want to build the compressed files or not. I'm pretty sure that for the playground we want to use the bootstrap logic instead.

@cpcallen How to handle local testing is blockly Eric, can you take a look?

File: .github/workflows/browser_test.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# This workflow will do a clean install, start the selenium server, and run
+# all of our browser based test
+
+name: Run browser test nightl
+
```
Comment (BeksOmega) [inline .github/workflows/browser_test.yml:5]: For security porpoises 🐳 can you also add a [permissions property](https://github.com/google/blockly/blob/9442b73d99fb63da4526c651ac41762e458b4ed7/.github/workflows/build.yml#L8)? I think it should just need read and not right.

File: .github/workflows/browser_test.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Linux Test Setup
+        if: runner.os == 'Linux'
+        run: source ./tests/scripts/setup_linux_env.sh
+
+      - name: MacOS Test Setup
+        if: runner.os == 'macOS'
+        run: source ./tests/scripts/setup_osx_env.sh
+
+      - name: Run
+        run: npx mocha ./tests/browser/test --config ./tests/browser/test/.mocharc.js
```
Comment (BeksOmega) [inline .github/workflows/browser_test.yml:52]: Can you add this command as a [package.json script](https://github.com/google/blockly/blob/9442b73d99fb63da4526c651ac41762e458b4ed7/package.json#L19) and run that instead? Good options might be `tests:selenium`, `tests:integration`, or anything of the form `tests:*`.

File: tests/browser/test/basic_block_factory_test.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
```
Comment (BeksOmega) [inline tests/browser/test/basic_block_factory_test.js:3]:
```suggestion
 * Copyright 2023 Google LLC
```

Here & below.

File: tests/browser/test/basic_block_factory_test.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * @fileoverview Node.js script to run Automated tests in Chrome, via webdriver.
+ */
+
+const webdriverio = require('webdriverio');
+const chai = require('chai');
+
+let browser;
+suite('Testing Connecting Blocks', function (done) {
+  this.timeout(0);
```
Comment (BeksOmega) [inline tests/browser/test/basic_block_factory_test.js:16]: Could you add a comment to this explaining why we're setting the timeout?

File: tests/browser/test/basic_block_factory_test.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  // Setup Selenium for each of the test
```
Comment (BeksOmega) [inline tests/browser/test/basic_block_factory_test.js:17]:
```suggestion
  // Setup Selenium for all of the tests.
```

I think this only runs once at the beginning of the suite, so I think this comment is clearer. But feel free to push back on this! (as well as any of my other comments :P )

File: tests/browser/test/basic_block_factory_test.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    // Use Selenium to bring up the page
+    const url =
+      'https://blockly-demo.appspot.com/static/demos/blockfactory/index.html';
+    console.log('Starting webdriverio...');
+    browser = await webdriverio.remote(options);
+    console.log('Loading URL: ' + url);
+    await browser.url(url);
+    return browser;
+  });
+  test('Testing Block Drag', async function () {
```
Comment (BeksOmega) [inline tests/browser/test/basic_block_factory_test.js:52]: As a formatting thing, can you add newlines between your different calls to `test` `suiteSetup` etc? I think it makes it easier to read (like adding newlines in between methods in a class).

File: .github/workflows/browser_test.yml [hunk 8]
```diff
@@ -0,0 +1,55 @@
+# This workflow will do a clean install, start the selenium server, and run
+# all of our browser based test
```
Comment (BeksOmega) [inline .github/workflows/browser_test.yml:2]:
```suggestion
# all of our browser based tests
```

**Review by BeksOmega — APPROVED** [review]

**Pull request merged**